proposals/zips/zip015.sol

```diff
@@ -0,0 +1,105 @@
+//SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.18;
+
+import {console} from "@forge-std/console.sol";
+
+import {Addresses} from "@proposals/Addresses.sol";
+import {Proposal} from "@proposals/proposalTypes/Proposal.sol";
+import {TimelockProposal} from "@proposals/proposalTypes/TimelockProposal.sol";
+
+import {Core} from "@protocol/core/Core.sol";
+import {Roles} from "@protocol/core/Roles.sol";
+import {Token} from "@protocol/token/Token.sol";
+import {CoreRef} from "@protocol/refs/CoreRef.sol";
+import {ERC1155Sale} from "@protocol/sale/ERC1155Sale.sol";
+
+contract zip015 is TimelockProposal {
+    Core private _core;
+
+    constructor() Proposal("ADMIN_TIMELOCK_CONTROLLER") {}
+
+    // Returns the name of the proposal.
+    function name() public pure override returns (string memory) {
+        return "ZIP015";
+    }
+
+    // Provides a brief description of the proposal.
+    function description() public pure override returns (string memory) {
+        return "The ZTX marketplace sale conracts";
+    }
+
+    function _beforeDeploy() internal override {
+        _core = Core(addresses.getAddress("CORE"));
+    }
+
+    function _build() internal override {
```

Comment:
only the `_afterDeploy()` calls were to be shifted to `_build()`. `_deploy()` would still be the same.